src/composables/PluginManager.js

```diff
@@ -199,19 +179,101 @@ export async function getFileInputs(plugin, fileInformations, projectName) {
 }
 
 /**
- * Update and draw new components.
- *
- * @param {Object} plugin - Contens components to update and draw.
- * @param {String} projectName - Project name.
+ * Init components.
+ * @param {String} defaultFolder - Default folder.
+ * @param {String} path - Model path (Plugin name & model name).
+ * @param {Object} plugin - Plugin corresponding to the model.
+ * @param {String} projectName - Name of the project.
+ * @return {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+export async function initComponents(defaultFolder, path, plugin, projectName) {
+  const dir = `${defaultFolder}${path}`;
+  const files = await readDir(`${projectName}/${dir}`);
+  const fileInformations = files.map((file) => new FileInformation({ path: `${dir}/${file}` }));
+
+  const fileInputs = getFileInputs(plugin, fileInformations, projectName);
+
+  const config = await readProjectFile(
+    projectName,
+    new FileInformation({
+      path: `${dir}/leto-modelizer.config.json`,
+    }),
+  );
+
+  plugin.parse(config, fileInputs);
+}
+
+/**
+ * Add a new component and render model.
+ * @param {String} defaultFolder - Default folder.
+ * @param {Object} definition - Definition of the component.
+ * @param {String} path - Model path (Plugin name & model name).
+ * @param {Object} plugin - Plugin corresponding to the model.
+ * @param {String} projectName - Name of the project.
+ * @return {Promise<void>} Promise with nothing on success otherwise an error.
  */
-export async function drawComponents(plugin, projectName) {
-  const fileInformations = await getProjectFiles(projectName);
+export async function addNewComponent(
+  defaultFolder,
```

Comment:
remove defaultFolder, it should be in th path

Reply:
done